### PR TITLE
chore: 🤖 add attach and detach storage policy permissions

### PR DESCRIPTION
**Pull request by DhariniJeeva**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12150

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12150)

## Description

This PR adds the required permissions in the scope model 

## How to Test

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- ~[ ] I have added before and after screenshots for UI changes~
- ~[ ] I have added JSON response output for API changes~
- ~[ ] I have added steps to reproduce and test for bug fixes in the description~
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/addon/abilities/scope.js
Comment (ZedLi) [inline addons/api/addon/abilities/scope.js:1]: Shouldn't this be in the storage policy ability and not the scope ability? Don't we normally keep it in the pertinent model and we would pass in the scope model?

File: ui/admin/app/abilities/scope.js [hunk 1]
```diff
@@ -47,4 +47,25 @@ export default class OverrideScopeAbility extends OverrideModelAbility {
         return super.canList;
     }
   }
+
+  /**
+   * Attaching a policy is allowed only if the feature flag is enabled
+   * @type {boolean}
+   */
+  get canAttachStoragePolicy() {
+    console.log(this, 'THIS');
```
Comment (ZedLi) [inline ui/admin/app/abilities/scope.js:56]: console log here

File: addons/api/addon/abilities/scope.js
Comment (DhariniJeeva) [reply addons/api/addon/abilities/scope.js:1]: hmm, I was looking at the target model and user model where we check for abilities in those directly, so I followed the same pattern, do you have any examples in mind for reference?

Comment (ZedLi) [reply addons/api/addon/abilities/scope.js:1]: For example, list or create permissions are also on the parent resource, which is often a scope but we use the base model for that. Create worker led permissions are still on the worker model and not the scope model

Comment (ZedLi) [reply addons/api/addon/abilities/scope.js:1]: But I also guess this is the first time we've had real permissions check that aren't collection actions on a scope, thinking again about it I suppose it makes sense?

I think the real inconsistency is how we handle authorized actions vs collection actions 🤷

**Review by ZedLi — APPROVED** [review]

**Pull request merged**